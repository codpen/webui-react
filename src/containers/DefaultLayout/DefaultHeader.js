import React, {Component} from 'react';
import {NavLink} from 'react-router-dom';
import {Badge, DropdownItem, DropdownMenu, DropdownToggle, Nav, NavItem} from 'reactstrap';
import PropTypes from 'prop-types';

import {AppAsideToggler, AppHeaderDropdown, AppNavbarBrand, AppSidebarToggler} from '@coreui/react';
import logo from '../../assets/img/brand/logo.png'
import favicon from '../../assets/img/brand/favicon.png'

const propTypes = {
    children: PropTypes.node,
};

const defaultProps = {};

class DefaultHeader extends Component {
    render() {

        // eslint-disable-next-line
        const {children, ...attributes} = this.props;

        return (
            <React.Fragment>
                <AppSidebarToggler className="d-lg-none" display="md" mobile/>
                <AppNavbarBrand
                    full={{src: logo, width: 89, height: 25, alt: 'CoreUI Logo'}}
                    minimized={{src: favicon, width: 30, height: 30, alt: 'CoreUI Logo'}}
                />
                <AppSidebarToggler className="d-md-down-none" display="lg"/>

                <Nav className="d-md-down-none" navbar>
                    <NavItem className="px-3">
                        <NavLink to="/dashboard" className="nav-link">Dashboard</NavLink>
                    </NavItem>

                </Nav>
                <Nav className="ml-auto" navbar>
                    <NavItem className="d-md-down-none">
                        <NavLink to="#" className="nav-link"><i className="icon-bell"></i><Badge pill
                                                                                                 color="danger">5</Badge></NavLink>
                    </NavItem>
                    <NavItem className="d-md-down-none">
                        <NavLink to="#" className="nav-link"><i className="icon-list"></i></NavLink>
                    </NavItem>
                    <NavItem className="d-md-down-none">
                        <NavLink to="#" className="nav-link"><i className="icon-location-pin"></i></NavLink>
                    </NavItem>
                    <AppHeaderDropdown direction="down">
                        <DropdownToggle nav>
                            <img src={'../../assets/img/avatars/6.jpg'} className="img-avatar"
                                 alt="admin@bootstrapmaster.com"/>
                        </DropdownToggle>
                        <DropdownMenu right style={{right: 'auto'}}>
                            <DropdownItem header tag="div"
                                          className="text-center"><strong>Account</strong></DropdownItem>
                            <DropdownItem><i className="fa fa-bell-o"></i> Updates<Badge
                                color="info">42</Badge></DropdownItem>

                        </DropdownMenu>
                    </AppHeaderDropdown>
                </Nav>
                <AppAsideToggler className="d-md-down-none"/>
                {/*<AppAsideToggler className="d-lg-none" mobile />*/}
            </React.Fragment>
        );
    }
}

DefaultHeader.propTypes = propTypes;
DefaultHeader.defaultProps = defaultProps;

export default DefaultHeader;
