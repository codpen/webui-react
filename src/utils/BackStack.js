import Stack from './Stack'
// A very simple stack implementation to handle back links in remote explorer
class BackStack {
    constructor() {
        this.backStack = new Stack();
        this.forwardStack = new Stack();
    }

    getLength() {
        return this.backStack.getLength();
    }

    push(item) {

        this.backStack.push(item);
        this.forwardStack.empty();

    }

    pop() {
        const temp = this.backStack.pop();
        return temp;
    }

    peek() {
        return this.backStack.peek();
    }

    empty() {
        this.backStack.empty();
        this.forwardStack.empty();
    }

    moveBack() {
        const temp = this.backStack.moveBack();
        if(temp)
            this.forwardStack.push(temp);
        return temp;
    }

    moveForward(){
        const temp = this.forwardStack.pop();
        if(temp){
            //Pop was successful
            this.backStack.push(temp);
        }
        return temp;

    }



}

export default BackStack;