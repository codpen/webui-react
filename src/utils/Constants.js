/**
 * Returns a string array with possible filtering options.
 * @type {string[]}
 */
export const visibilityFilteringOptions = ["Images", "Pdf", "Videos"];

/**
 * Refresh timeout for checking backend status.
 * In milliseconds.
 * @type {number}
 */
export const STATUS_REFRESH_TIMEOUT = 1000 * 5;
/**
 * Refresh timeout for checking if the config is created.
 * In milliseconds.
 * @type {number}
 */
export const NEW_DRIVE_CONFIG_REFRESH_TIMEOUT = 1000;
export const MODAL_ROOT_ELEMENT = "modal-root";

export const USER_NAME_KEY = "userName";
export const PASSWORD_KEY = "password";
export const IP_ADDRESS_KEY = "ipAddress";

export const FILES_VIEW_HEIGHT = "75vh";