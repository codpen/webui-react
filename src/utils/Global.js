global.ipAddress = "http://127.0.0.1:5572";