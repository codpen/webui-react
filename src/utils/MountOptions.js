export const vfsOptions = {
    "CacheMaxAge": {
        Type: 'int',
        Default: 3600000000000
    },
    "CacheMaxSize": {
        Type: 'int',
        Default: -1,
        Help: ''
    },
    "CacheMode": {
        Type: 'options',
        Options: [
            {value:'off', key: 0},
            {value:'minimal', key: 1},
            {value:'writes', key: 2},
            {value:'full', key: 3},
        ],
        Default: 0,
        Help: ''
    },
    "CachePollInterval": {
        Type: 'int',
        Default: 60000000000,
        Help: ''
    },
    "CaseInsensitive": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "ChunkSize":{
        Type: 'int',
        Default: 134217728,
        Help: ''
    },
    "ChunkSizeLimit": {
        Type: 'int',
        Default: -1,
        Help: ''
    },
    "DirCacheTime": {
        Type: 'int',
        Default: 300000000000,
        Help: ''
    },
    "DirPerms": {
        Type: 'int',
        Default: 511,
        Help: ''
    },
    "FilePerms": {
        Type: 'int',
        Default: 438,
        Help: ''
    },
    "GID": {
        Type: 'int',
        Default: 1000,
        Help: ''
    },
    "NoChecksum": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "NoModTime": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "NoSeek": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "PollInterval": {
        Type: 'int',
        Default: 60000000000,
        Help: ''
    },
    "ReadAhead": {
        Type: 'int',
        Default: 0,
        Help: ''
    },
    "ReadOnly": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "ReadWait": {
        Type: 'int',
        Default: 20000000,
        Help: ''
    },
    "UID": {
        Type: 'int',
        Default: 1000,
        Help: ''
    },
    "Umask": {
        Type: 'int',
        Default: 2,
        Help: ''
    },
    "WriteBack": {
        Type: 'int',
        Default: 5000000000,
        Help: ''
    },
    "WriteWait": {
        Type: 'int',
        Default: 1000000000,
        Help: ''
    }
}

export const mountOptions = {
    "AllowNonEmpty": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "AllowOther": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "AllowRoot": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "AsyncRead": {
        Type: 'bool',
        Default: true,
        Help: ''
    },
    "AttrTimeout": {
        Type: 'int',
        Default: 1000000000,
        Help: ''
    },
    "Daemon": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "DaemonTimeout": {
        Type: 'int',
        Default: 0,
        Help: ''
    },
    "DebugFUSE": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "DefaultPermissions": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "ExtraFlags": {
        Type: 'array',
        Default: [],
        Help: ''
    },
    "ExtraOptions": {
        Type: 'array',
        Default: [],
        Help: ''
    },
    "MaxReadAhead": {
        Type: 'int',
        Default: 131072,
        Help: ''
    },
    "NoAppleDouble": {
        Type: 'bool',
        Default: true,
        Help: ''
    },
    "NoAppleXattr": {
        Type: 'bool',
        Default: false,
        Help: ''
    },
    "VolumeName": {
        Type: 'string',
        Default: "",
        Help: ''
    },
    "WritebackCache": {
        Type: 'bool',
        Default: false,
        Help: ''
    }
}