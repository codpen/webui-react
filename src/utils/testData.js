export const TEST_REDUX_PROPS = {
    status: {
        isConnected: true,
        jobs: {
            bytes: 0,
            checks: 0,
            deletes: 0,
            elapsedTime: 737.732988329,
            errors: 0,
            fatalError: false,
            retryError: false,
            speed: 0,
            transfers: 0
        },
        speed: [
            {
                elapsedTime: 737.732988329,
                speed: 0,
                speedAvg: 0
            }
        ],
        runningAvgSpeed: 0,
        checkStatus: false,
        bandwidth: {
            bytesPerSecond: -1,
            rate: 'off'
        }
    },
    config: {
        providers: [{
            Name: 'b2',
            Description: 'Backblaze B2',
            Prefix: 'b2',
            Options: [
                {
                    Name: 'account',
                    Help: 'Account ID or Application Key ID',
                    Provider: '',
                    Default: '',
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: true,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: false,
                    DefaultStr: '',
                    ValueStr: '',
                    Type: 'string'
                },
                {
                    Name: 'key',
                    Help: 'Application Key',
                    Provider: '',
                    Default: '',
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: true,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: false,
                    DefaultStr: '',
                    ValueStr: '',
                    Type: 'string'
                },
                {
                    Name: 'endpoint',
                    Help: 'Endpoint for the service.\nLeave blank normally.',
                    Provider: '',
                    Default: '',
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: '',
                    ValueStr: '',
                    Type: 'string'
                },
                {
                    Name: 'test_mode',
                    Help: 'A flag string for X-Bz-Test-Mode header for debugging.\n\nThis is for debugging purposes only. Setting it to one of the strings\nbelow will cause b2 to return specific errors:\n\n  * "fail_some_uploads"\n  * "expire_some_account_authorization_tokens"\n  * "force_cap_exceeded"\n\nThese will be set in the "X-Bz-Test-Mode" header which is documented\nin the [b2 integrations checklist](https://www.backblaze.com/b2/docs/integration_checklist.html).',
                    Provider: '',
                    Default: '',
                    Value: null,
                    ShortOpt: '',
                    Hide: 2,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: '',
                    ValueStr: '',
                    Type: 'string'
                },
                {
                    Name: 'versions',
                    Help: 'Include old versions in directory listings.\nNote that when using this no file write operations are permitted,\nso you can\'t upload files or delete them.',
                    Provider: '',
                    Default: false,
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: 'false',
                    ValueStr: 'false',
                    Type: 'bool'
                },
                {
                    Name: 'hard_delete',
                    Help: 'Permanently delete files on remote removal, otherwise hide files.',
                    Provider: '',
                    Default: false,
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: false,
                    DefaultStr: 'false',
                    ValueStr: 'false',
                    Type: 'bool'
                },
                {
                    Name: 'upload_cutoff',
                    Help: 'Cutoff for switching to chunked upload.\n\nFiles above this size will be uploaded in chunks of "--b2-chunk-size".\n\nThis value should be set no larger than 4.657GiB (== 5GB).',
                    Provider: '',
                    Default: 209715200,
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: '200M',
                    ValueStr: '200M',
                    Type: 'SizeSuffix'
                },
                {
                    Name: 'chunk_size',
                    Help: 'Upload chunk size. Must fit in memory.\n\nWhen uploading large files, chunk the file into this size.  Note that\nthese chunks are buffered in memory and there might a maximum of\n"--transfers" chunks in progress at once.  5,000,000 Bytes is the\nminimum size.',
                    Provider: '',
                    Default: 100663296,
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: '96M',
                    ValueStr: '96M',
                    Type: 'SizeSuffix'
                },
                {
                    Name: 'disable_checksum',
                    Help: 'Disable checksums for large (> upload cutoff) files',
                    Provider: '',
                    Default: false,
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: 'false',
                    ValueStr: 'false',
                    Type: 'bool'
                },
                {
                    Name: 'download_url',
                    Help: 'Custom endpoint for downloads.\n\nThis is usually set to a Cloudflare CDN URL as Backblaze offers\nfree egress for data downloaded through the Cloudflare network.\nThis is probably only useful for a public bucket.\nLeave blank if you want to use the endpoint provided by Backblaze.',
                    Provider: '',
                    Default: '',
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: '',
                    ValueStr: '',
                    Type: 'string'
                },
                {
                    Name: 'download_auth_duration',
                    Help: 'Time before the authorization token will expire in s or suffix ms|s|m|h|d.\n\nThe duration before the download authorization token will expire.\nThe minimum value is 1 second. The maximum value is one week.',
                    Provider: '',
                    Default: 604800000000000,
                    Value: null,
                    ShortOpt: '',
                    Hide: 0,
                    Required: false,
                    IsPassword: false,
                    NoPrefix: false,
                    Advanced: true,
                    DefaultStr: '1w',
                    ValueStr: '1w',
                    Type: 'Duration'
                }
            ]
        }],
        configDump: {
            local: {
                type: 'local'
            },
            mydrive: {
                token: '',
                type: 'drive'
            }
        },
        hasError: false
    },
    remote: {
        configs: {
            local: {
                Features: {
                    About: true,
                    BucketBased: false,
                    CanHaveEmptyDirectories: true,
                    CaseInsensitive: false,
                    ChangeNotify: false,
                    CleanUp: false,
                    Copy: false,
                    DirCacheFlush: false,
                    DirMove: true,
                    DuplicateFiles: false,
                    GetTier: false,
                    ListR: false,
                    MergeDirs: false,
                    Move: true,
                    OpenWriterAt: true,
                    PublicLink: false,
                    Purge: true,
                    PutStream: true,
                    PutUnchecked: false,
                    ReadMimeType: false,
                    ServerSideAcrossConfigs: false,
                    SetTier: false,
                    SetWrapper: false,
                    UnWrap: false,
                    WrapFs: false,
                    WriteMimeType: false
                },
                Hashes: [
                    'MD5',
                    'SHA-1',
                    'DropboxHash',
                    'QuickXorHash',
                    'Whirlpool'
                ],
                Name: 'local',
                Precision: 1,
                Root: '/home/negative0/Projects/rclone/rclone',
                String: 'Local file system at /home/negative0/Projects/rclone/rclone'
            }
        },
        remotes: [
            'local',
            'mydrive'
        ],
        files: {
            'local-': {
                time: '2019-07-13T07:43:19.196Z',
                files: [
                    {
                        Path: '.appveyor.yml',
                        Name: '.appveyor.yml',
                        Size: 1434,
                        MimeType: 'application/octet-stream',
                        ModTime: '2019-06-07T22:44:37.357539000+05:30',
                        IsDir: false
                    },
                    {
                        Path: '.circleci',
                        Name: '.circleci',
                        Size: -1,
                        MimeType: 'inode/directory',
                        ModTime: '2019-06-24T09:45:29.483620000+05:30',
                        IsDir: true
                    }
                ]
            }
        },
        hasError: false,
        numCols: 1,
        distractionFreeMode: false
    },
    explorer: {
        backStacks: {
            '0': {
                backStack: {
                    items: [
                        {
                            remoteName: 'local',
                            remotePath: ''
                        }
                    ],
                    count: 1
                },
                forwardStack: {
                    items: [],
                    count: 0
                }
            }
        },
        currentPaths: {
            '0': {
                remoteName: 'local',
                remotePath: ''
            }
        },
        visibilityFilters: {},
        gridMode: {},
        searchQueries: {
            '0': ''
        },
        loadImages: {
            '0': false
        }
    },
    providerStatus: {
        about: {}
    },
    user: {
        auth: {
            userName: 'chaitanya',
            password: 'abcd',
            ipAddress: 'http://localhost:5572/',
            interceptor: 2
        }
    },
    imageLoader: {}
};

export const TEST_FILE_CONTAINER_ID = '0'

export const FILE_ITEM_DATA = {
    Path: '.appveyor.yml',
    Name: '.appveyor.yml',
    Size: 1434,
    MimeType: 'application/octet-stream',
    ModTime: '2019-06-07T22:44:37.357539000+05:30',
    IsDir: false
};

export const DIR_ITEM_DATA = {
    Path: 'abc',
    Name: 'abc',
    Size: 0,
    MimeType: '',
    ModTime: '2019-06-07T22:44:37.357539000+05:30',
    IsDir: true
};