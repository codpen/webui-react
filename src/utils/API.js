import axios from "axios";
import './Global'

let axiosInstance = axios.create({
    baseURL: global.ipAddress,
    headers: {'Content-Type': 'application/json'},
    responseType: "json"
});

export function performMoveFile(srcFs, srcRemote, dstFs, dstRemote, Name, IsDir) {
    return performCopyOrMoveFile(srcFs, srcRemote, dstFs, dstRemote, Name, IsDir, 'move');
}

export function performCopyFile(srcFs, srcRemote, dstFs, dstRemote, Name, IsDir) {

    return performCopyOrMoveFile(srcFs, srcRemote, dstFs, dstRemote, Name, IsDir, 'copy');
}

async function performCopyOrMoveFile(srcFs, srcRemote, dstFs, dstRemote, Name, IsDir, mode) {
    let url = "";
    if (mode === "move") {
        url = "operations/movefile";
    } else {
        url = "/operations/copyfile";
    }
    if (IsDir) {
        // if (dstRemote === "") {
        //     dstRemote = Name;
        // } else {
        //     dstRemote += "/" + Name;
        // }
        // const data = {
        //     srcFs: srcFs + srcRemote,
        //     dstFs: dstFs + dstRemote
        // };

        // let res = await axiosInstance.post("/operations/copy")
        // let res = await axiosInstance.post("/sync/copy", data);
        // console.log("Dir copy Res", res);
        // this.setState({isOperationInProgress: false})
        console.log("Copying directory feature is not yet implemented");

    } else {
        if (dstRemote === "") {
            dstRemote = Name;
        } else {
            dstRemote += "/" + Name;
        }

        const data = {
            srcFs: srcFs,
            srcRemote: srcRemote,
            dstFs: dstFs,
            dstRemote: dstRemote,
        };
        let res = await axiosInstance.post(url, data);
        console.log("Res", res);

    }
}

export default axiosInstance;
