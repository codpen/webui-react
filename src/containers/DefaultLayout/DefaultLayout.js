import React, {Component, Suspense} from 'react';
import {Redirect, Route, Switch} from 'react-router-dom';
import {Container} from 'reactstrap';
import {getVersion} from "../../actions/versionActions";

import {
    AppBreadcrumb,
    AppFooter,
    AppHeader,
    AppSidebar,
    AppSidebarFooter,
    AppSidebarForm,
    AppSidebarHeader,
    AppSidebarMinimizer,
    AppSidebarNav,
} from '@coreui/react';
// sidebar nav config
import navigation from '../../_nav';
// routes config
import routes from '../../routes';
import {connect} from "react-redux";
import {AUTH_KEY} from "../../utils/Constants";
import ErrorBoundary from "../../ErrorHandling/ErrorBoundary";

// const DefaultAside = React.lazy(() => import('./DefaultAside'));
const DefaultFooter = React.lazy(() => import('./DefaultFooter'));
const DefaultHeader = React.lazy(() => import('./DefaultHeader'));

const VERSION_NAV_ITEM_ATTRS = {
    attributes: { target: '_blank' },
    class: 'mt-auto',
    icon: 'cui-cog',
    url: 'https://rclone.org/changelog',
    variant: 'success'
}
class DefaultLayout extends Component {

    loading = () => <div className="animated fadeIn pt-1 text-center">Loading...</div>;

    get navConfig() {
        return {
            items: [
                ...navigation.items,
                {
                    name: this.props.version.version,
                    ...VERSION_NAV_ITEM_ATTRS
                }
            ]
        }
    }

    componentWillMount() {
        if (!localStorage.getItem(AUTH_KEY)) {
            this.props.history.push('/login');
        } else {
            this.props.getVersion();
        }
    }

    render() {
        // console.log("isConnected, default layout", this.props.isConnected);
        return (


            <div className="app" data-test="defaultLayout">
                <ErrorBoundary>
                    <AppHeader fixed>
                        <Suspense fallback={this.loading()}>
                            <DefaultHeader onLogout={e => this.signOut(e)}/>
                        </Suspense>
                    </AppHeader>
                    <div className="app-body">
                        <AppSidebar fixed display="lg">
                            <AppSidebarHeader/>
                            <AppSidebarForm/>
                            <Suspense fallback={this.loading()}>
                                <AppSidebarNav navConfig={this.navConfig} />
                            </Suspense>
                            <AppSidebarFooter/>
                            <AppSidebarMinimizer/>
                        </AppSidebar>
                        <main className="main">
                            <AppBreadcrumb appRoutes={routes}/>
                            <Container fluid>
                                <Suspense fallback={this.loading()}>
                                    <Switch>
                                        {
                                            routes.map((route, idx) => {
                                                return route.component ? (
                                                    <Route
                                                        key={idx}
                                                        path={route.path}
                                                        exact={route.exact}
                                                        name={route.name}
                                                        render={props => (
                                                            <route.component {...props} />
                                                        )}/>
                                                ) : (null);
                                            })
                                        }
                                        <Redirect from="/" to="/login"/>
                                    </Switch>
                                </Suspense>
                            </Container>
                        </main>
                    </div>
                    <AppFooter>
                        <Suspense fallback={this.loading()}>
                            <DefaultFooter/>
                        </Suspense>
                    </AppFooter>
                </ErrorBoundary>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    isConnected: state.status.isConnected,
    version: state.version,
});

export default connect(mapStateToProps, { getVersion })(DefaultLayout);
