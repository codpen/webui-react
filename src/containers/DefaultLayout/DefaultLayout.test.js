import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import DefaultLayout from "./DefaultLayout";
import {TEST_REDUX_PROPS} from "../../utils/testData";
import {findByTestAttr, testStore} from "../../../Utils";

const setUp = (initialState = {}, props = {}) => {
    const store = testStore(initialState);
    const component = shallow(<DefaultLayout {...props} store={store}/>);
    return component.childAt(0).dive();
};


describe('Remote Explorer', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {
                ...TEST_REDUX_PROPS
            };


            const props = {

                history: {
                    push: jest.fn()
                }

            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "defaultLayout");
            expect(component).toHaveLength(1);
        });
        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });
    });

});