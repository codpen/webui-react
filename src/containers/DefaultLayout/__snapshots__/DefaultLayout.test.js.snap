// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remote Explorer renders should match snapshot 1`] = `
<div
  className="app"
  data-test="defaultLayout"
>
  <ErrorBoundary>
    <AppHeader
      fixed={true}
      tag="header"
    >
      <Suspense
        fallback={
          <div
            className="animated fadeIn pt-1 text-center"
          >
            Loading...
          </div>
        }
      >
        <lazy
          onLogout={[Function]}
        />
      </Suspense>
    </AppHeader>
    <div
      className="app-body"
    >
      <AppSidebar
        compact={false}
        display="lg"
        fixed={true}
        isOpen={false}
        minimized={false}
        offCanvas={false}
        tag="div"
      >
        <AppSidebarHeader
          tag="div"
        />
        <AppSidebarForm
          tag="div"
        />
        <Suspense
          fallback={
            <div
              className="animated fadeIn pt-1 text-center"
            >
              Loading...
            </div>
          }
        >
          <AppSidebarNav
            history={
              Object {
                "push": [MockFunction] {
                  "calls": Array [
                    Array [
                      "/login",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              }
            }
            isConnected={true}
            isOpen={false}
            navConfig={
              Object {
                "items": Array [
                  Object {
                    "icon": "icon-speedometer",
                    "name": "Dashboard",
                    "url": "/dashboard",
                  },
                  Object {
                    "icon": "icon-note",
                    "name": "Configs",
                    "url": "/showconfig",
                  },
                  Object {
                    "icon": "icon-screen-desktop",
                    "name": "Explorer",
                    "url": "/remoteExplorer",
                  },
                  Object {
                    "icon": "icon-star",
                    "name": "Backend",
                    "url": "/rcloneBackend",
                  },
                  Object {
                    "icon": "icon-logout",
                    "name": "Log Out",
                    "url": "/login",
                  },
                ],
              }
            }
            store={
              Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              }
            }
            tag="nav"
          />
        </Suspense>
        <AppSidebarFooter
          tag="div"
        />
        <AppSidebarMinimizer
          tag="button"
          type="button"
        />
      </AppSidebar>
      <main
        className="main"
      >
        <AppBreadcrumb
          appRoutes={
            Array [
              Object {
                "exact": true,
                "name": "Home",
                "path": "/",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "name": "New Drive",
                "path": "/newdrive/edit/:drivePrefix",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "exact": true,
                "name": "New Drive",
                "path": "/newdrive",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "exact": true,
                "name": "New Drive",
                "path": "/login",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "name": "Dashboard",
                "path": "/dashboard",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "name": "Configs",
                "path": "/showconfig",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "exact": true,
                "name": "Explorer",
                "path": "/remoteExplorer/:remoteName/:remotePath",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "name": "Explorer",
                "path": "/remoteExplorer",
              },
              Object {
                "component": Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                },
                "name": "Rclone Backend",
                "path": "/rcloneBackend",
              },
            ]
          }
          className=""
          tag="div"
        />
        <Container
          fluid={true}
          tag="div"
        >
          <Suspense
            fallback={
              <div
                className="animated fadeIn pt-1 text-center"
              >
                Loading...
              </div>
            }
          >
            <Switch>
              <Route
                key="1"
                name="New Drive"
                path="/newdrive/edit/:drivePrefix"
                render={[Function]}
              />
              <Route
                exact={true}
                key="2"
                name="New Drive"
                path="/newdrive"
                render={[Function]}
              />
              <Route
                exact={true}
                key="3"
                name="New Drive"
                path="/login"
                render={[Function]}
              />
              <Route
                key="4"
                name="Dashboard"
                path="/dashboard"
                render={[Function]}
              />
              <Route
                key="5"
                name="Configs"
                path="/showconfig"
                render={[Function]}
              />
              <Route
                exact={true}
                key="6"
                name="Explorer"
                path="/remoteExplorer/:remoteName/:remotePath"
                render={[Function]}
              />
              <Route
                key="7"
                name="Explorer"
                path="/remoteExplorer"
                render={[Function]}
              />
              <Route
                key="8"
                name="Rclone Backend"
                path="/rcloneBackend"
                render={[Function]}
              />
              <Redirect
                from="/"
                to="/login"
              />
            </Switch>
          </Suspense>
        </Container>
      </main>
      <AppAside
        display=""
        fixed={true}
        isOpen={false}
        offCanvas={true}
        tag="aside"
      >
        <Suspense
          fallback={
            <div
              className="animated fadeIn pt-1 text-center"
            >
              Loading...
            </div>
          }
        >
          <lazy />
        </Suspense>
      </AppAside>
    </div>
    <AppFooter
      fixed={false}
      tag="footer"
    >
      <Suspense
        fallback={
          <div
            className="animated fadeIn pt-1 text-center"
          >
            Loading...
          </div>
        }
      >
        <lazy />
      </Suspense>
    </AppFooter>
  </ErrorBoundary>
</div>
`;
